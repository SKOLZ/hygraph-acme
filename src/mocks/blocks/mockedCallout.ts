export const mockedCallout = {
  __typename: "Callout" as const,
  id: "clt7bqmp49dkq07uj23atb9h5",
  title: "Start your 30-day free trial",
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "Join over 4,000+ startups already growing with us.",
            },
          ],
        },
      ],
    },
  },
  actionButton: {
    buttonText: "Get started",
    size: null,
  },
};
