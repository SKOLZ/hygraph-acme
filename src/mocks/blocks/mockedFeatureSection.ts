import { BlockFeatureSection, Direction, Sizes } from "@/generated/graphql";

export const mockedFeatureSection = {
  __typename: "FeatureSection" as const,
  id: "clt7bqmp09dkf07uj2k6in82a",
  label: "feature section label",
  title: "feature section title",
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "As a composable and headless platform, Acme works with all your preferred frontend frameworks, existing systems, GraphQL and REST APIs.",
            },
          ],
        },
      ],
    },
  },
  media: {
    __typename: "StaticImage" as const,
    image: {
      url: "https://media.graphassets.com/bbd3PKNcTBGVkNbcois7",
      width: 480,
      height: 320,
    },
    altText: "Stats page showing a profile views chart and a chat window",
  },
  direction: Direction.LeftToRight,
  featureSectionButton: {
    buttonText: "Get started",
    size: Sizes.Lg,
  },
};
