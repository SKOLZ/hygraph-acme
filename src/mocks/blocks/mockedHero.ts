import { BlockHero, Sizes } from "@/generated/graphql";

export const mockedHero = {
  __typename: "Hero" as const,
  id: "1",
  title: `Hero title`,
  description: {
    raw: {
      children: [
        {
          type: "paragraph",
          children: [
            {
              text: "Hero description",
            },
          ],
        },
      ],
    },
  },
  buttonActions: [
    {
      id: "clt7bqmoz9dkc07ujrs01ghn3",
      buttonText: "Get started",
      size: Sizes.Lg,
    },
  ],
};
