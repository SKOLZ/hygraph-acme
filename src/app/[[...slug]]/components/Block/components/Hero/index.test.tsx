import { render } from "@testing-library/react"
import { Hero } from "."
import { mockedHero } from "@/mocks/blocks/mockedHero";

describe("Hero", () => {
  it("renders the component correctly", () => {
    const { container } = render(<Hero {...mockedHero} />);
    expect(container).toMatchSnapshot();
  })
})
