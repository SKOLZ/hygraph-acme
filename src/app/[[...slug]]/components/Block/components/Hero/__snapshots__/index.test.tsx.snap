// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders the component correctly 1`] = `
<div>
  <div
    class="mx-auto w-full flex max-w-screen-xl flex-col items-center px-8 text-center"
  >
    <h1
      class="title-1 mb-4 text-secondary-900 sm:mb-6"
    >
      Hero title
    </h1>
    <p
      class="text-2 mb-8 max-w-3xl text-secondary-400 sm:mb-12"
    >
      Hero description
    </p>
    <div
      class="flex flex-col items-center justify-center gap-3 self-stretch sm:flex-row"
    >
      <a
        class="button button-lg "
        href="#"
      >
        Get started
      </a>
    </div>
  </div>
</div>
`;
